ts/src/main/scala/ch/epfl/ts/indicators/RangeIndicator.scala

```diff
@@ -0,0 +1,77 @@
+package ch.epfl.ts.indicators
+
+import ch.epfl.ts.component.Component
+import ch.epfl.ts.data.OHLC
+import scala.collection.mutable.MutableList
+import com.sun.xml.internal.bind.v2.model.annotation.Init
+
+class RI(val support : Double, val resistance : Double, val period : Int)
+case class RI2( override val support : Double, override val resistance : Double, override val period : Int ) extends RI(support,resistance,period)
+/**
+ * This indicator is meant to draw support and resistance range over a given time period and with some given tolerance 
+ * Note : not always possible to retrieve an accurate range over large period if there is lot of same prices that appears !
+ * @param tolerance represent the number of points that are allowed to be above resp. below the resistance resp. support line 
+ */
+class RangeIndicator(timePeriod : Int, tolerance : Int) extends Component {
+  
+  // resistance will be the period^th highest prices  
+  var support : Double = 0.0
+  
+   // support will be the period^th lowest prices
+  var resistance : Double = 0.0
+  
+  var price : Double = 0.0 
+ 
+  var pricePeriod : MutableList[Double] = MutableList[Double]()
```

Comment:
This variable's name is not very clear. "pricePeriod" (singular) suggests it is on thing, but it turns out to be a list. Maybe `priceInPeriod`?